code/modules/admin/verbs/map_export.dm

```diff
@@ -0,0 +1,261 @@
+/client/proc/map_export()
+	set category = "Debug"
+	set name = "Export Map"
+	set desc = "Select a part of the map by coordinates and download it."
+	
+	if(!check_rights(R_DEBUG))
+		return
+
+	var/z_level = input("Export Which Z-Level?", "Map Exporter", 2) as num
+	var/start_x = input("Start X?", "Map Exporter", 1) as num
+	var/start_y = input("Start Y?", "Map Exporter", 1) as num
+	var/end_x = input("End X?", "Map Exporter", world.maxx-1) as num
+	var/end_y = input("End Y?", "Map Exporter", world.maxy-1) as num
+	var/date = time2text(world.timeofday, "YYYY-MM-DD_hh-mm-ss")
+	var/file_name = sanitize_filename(input("Filename?", "Map Exporter", "exportedmap_[date]") as text)
+	var/confirm = tgui_alert(usr, "Are you sure you want to do this? This will cause extreme lag!", "Map Exporter", list("Yes", "No"))
+
+	if(confirm != "Yes")
+		return
+
+	var map_text = write_map(start_x, start_y, z_level, end_x, end_y, z_level)
+	text2file(map_text, "data/[file_name].dmm")
+	usr << ftp("data/[file_name].dmm", "[file_name].dmm")
+
+/proc/sanitize_filename(t)
```

Comment:
do we not have something like this already? regardless, no new single-letter variables.

Reply:
don't think I've found any)